lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -104,6 +104,12 @@ def exec_migration(conn, direction)
 
   def safely_acquire_lock_for_table(table, &block)
     _check_postgres_adapter!
+
+    # This method is NOT safe to run more than once in a DDL transaction,
+    # as it could potentially lead to multiple tables being locked
+    # indefinitely
+    return block.call unless disable_ddl_transaction
```

Comment:
I'm not sure this is correct; if we allow DDL transactionality then the locking problem exists whether or not we use explicit locking with this method or the locks automatically acquired by any DDL statement in Postgres.

Because it's not possible to avoid locks altogether, allowing DDL transactionality across multiple statements (single statements are already inherently transactional) necessarily means that you have to accept the risk of a serious outage (because of the possibility of deadlocks, etc.)

The one exception to this is when multiple changes need to be made against the same object, but in that case I think the appropriate action is to wrap that set of changes in an explicit `transaction do; ...; end` block.